src/Employer/UnitTests/Employer.Web/ViewModels/Part1/Location/LocationEditViewModelTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using Esfa.Recruit.Employer.Web.ViewModels.Location;
+using FluentAssertions;
+using Xunit;
+
+namespace Esfa.Recruit.Employer.UnitTests.Employer.Web.ViewModels.Part1.Location
+{
+    public class LocationEditViewModelTests
+    {
+        [Fact]
+        public void ShouldErrorIfPostcodeNotSupplied()
+        {
+            var m = new LocationEditModel
+            {
+                Postcode = null,
+                EmployerAccountId = null
+            };
+
+            var context = new ValidationContext(m, null, null);
+            var result = new List<ValidationResult>();
+
+            // Act
+            var isValid = Validator.TryValidateObject(m, context, result, true);
+
+            isValid.Should().BeFalse();
+            result.Should().HaveCount(3);
+            result.Single(r => r.MemberNames.Single() == "VacancyId").ErrorMessage.Should().Be("The field VacancyId is invalid.");
+            result.Single(r => r.MemberNames.Single() == "Postcode").ErrorMessage.Should().Be("The Postcode field is required.");
+            result.Single(r => r.MemberNames.Single() == "EmployerAccountId").ErrorMessage.Should().Be("The EmployerAccountId field is required.");
+        }
+
+        [Fact]
+        public void ShouldErrorIfPostcodeInvalid()
+        {
+            var m = new LocationEditModel
+            {
+                Postcode = "invalid post code",
+                EmployerAccountId = "employer account Id",
+                VacancyId = Guid.NewGuid()
+            };
+
+            var context = new ValidationContext(m, null, null);
+            var result = new List<ValidationResult>();
+
+            // Act
+            var isValid = Validator.TryValidateObject(m, context, result, true);
+
+            isValid.Should().BeFalse();
+            result.Should().HaveCount(1);
+            result.Single(r => r.MemberNames.Single() == "Postcode").ErrorMessage.Should().Be("The Postcode is not valid.");

```

Comment:
I think the default validation mechanism outputs messages such as `The postcode *field* is required`. Maybe we should use the same format for our own custom error messages.

Reply:
Yeah. My plan is to revisit the all validations once the screens are done.

Reply:
Cool, I will add it to the tech debt page just as a reminder. We will need a holistic review of the whole validation/messages thing anyway.